### PR TITLE
Add audit fields to `moped_proj_phases`

### PR DESCRIPTION
**Pull request by johnclary**
## Associated issues
- https://github.com/cityofaustin/atd-data-tech/issues/15327

## Testing
**URL to test:** Local

You will use the Moped Editor and a SQL client to inspect the values of the audit fields on `moped_proj_phases` and `moped_project`.

**✨  Bonus points for using prod data ✨**
1. Add a new phase to a project. Verify that the `created_at`, `updated_at`, `created_by_user_id`, and `updated_by_user_id` fields are all populated as expected. On the `moped_project` record, verify that `updated_at` and `updated_by_user_id` values match the phase record you created.
2. Logout and login as different user (e.g. switch from the Test Admin to the Test Editor).
3. Edit the phase you created. Verify that the  `updated_at` and `updated_by_user_id` fields are all populated as expected. On the `moped_project` record, verify that `updated_at` and `updated_by_user_id` values match the phase record you edited.
4. Inspect the activity log to verify your changes are being captured, bearing in mind this [known bug](https://github.com/cityofaustin/atd-data-tech/issues/15397).
5. Head over to the **Funding** tab and make sure you can add and edit funding records without issue. Although you will need to [fill in all fields](https://github.com/cityofaustin/atd-data-tech/issues/15314).
6. Run the down migrations: `hasura migrate apply --down 2`
---
#### Ship list
- [ ] Code reviewed 
- [ ] Product manager approved
- [ ] Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

### CODE REVIEW

File: moped-database/metadata/tables.yaml [hunk 1]
```diff
@@ -3288,56 +3292,66 @@
...
           - subphase_id
-          - is_phase_start_confirmed
-          - is_phase_end_confirmed
+          - phase_description
+          - created_at
+          - phase_end
+          - phase_start
+          - updated_by_user_id
+          - is_deleted
+          - updated_at
```
Comment (johnclary) [inline moped-database/metadata/tables.yaml:3309]: the column permission changes are:
- remove `date_added`
- add: created_at, updated_at, created_by_user_id, updated_by_user_id
- configure `set` rules for `created_by_user_id` (insert only) and updated_by_user_id (insert and update)

File: moped-database/migrations/1706213015666_add_user_id_fks_to_audit_fields/up.sql
Comment (johnclary) [inline moped-database/migrations/1706213015666_add_user_id_fks_to_audit_fields/up.sql:1]: It occurs to me that we are not configuring these relationships in the Hasura graphql engine, which is ok for now, but we definitely want to make sure we're enforcing these constraints 🔒

File: moped-database/seeds/1602292389297_initial_seed_staging.sql [hunk 2]
```diff
@@ -67,9 +67,9 @@ INSERT INTO public.moped_proj_work_activity (id, workgroup_contractor, contract_
...
+INSERT INTO public.moped_proj_funding (proj_funding_id, project_id, created_at, created_by_user_id, funding_source_id, funding_program_id, funding_amount, funding_description, funding_status_id, fund, dept_unit, is_deleted) VALUES (4, 227, '2022-11-12 18:12:47.295689+00', 1, 9, 11, 0, 'This is funding description', 1, '{"fund_id": "4730", "fund_name": "PARKING CIP"}', '{"dept": "2507", "unit": "9040", "dept_id": "180775091", "dept_unit_id": "207413510", "unit_long_name": "Parking Initiatives", "unit_short_name": "8400-2507", "dept_unit_status": "Active"}', false);
+INSERT INTO public.moped_proj_funding (proj_funding_id, project_id, created_at, created_by_user_id, funding_source_id, funding_program_id, funding_amount, funding_description, funding_status_id, fund, dept_unit, is_deleted) VALUES (3, 229, '2022-11-12 18:12:47.295683+00', 1, NULL, NULL, NULL, NULL, 2, '{"fund_id": "4730", "fund_name": "PARKING CIP"}', '{"dept": "2507", "unit": "9042", "dept_id": "180775091", "dept_unit_id": "153705858", "unit_long_name": "New & Repair Parking Stations", "unit_short_name": "4730-2507", "dept_unit_status": "Active"}', false);
```
Comment (johnclary) [inline moped-database/seeds/1602292389297_initial_seed_staging.sql:72]: i changed `created_by_user_id` value from `10` to `1` bc we are enforcing the foreign key constraint now.

**Review by chiaberry — APPROVED** [review]
Got my bonus points, used prod data and different user accounts

✅

**Comment by johnclary** [comment]
I'm going to merge this in the interest of avoiding metadata conflicts, but am happy to open a a follow-up PR with revisions if needed.

**Pull request merged**